### ISSUE TITLE
Calc: Cannot edit email address

### ISSUE THREAD

**Issue by relikd** [CollaboraOnline/online#7045]
### Describe the Bug

After inserting (or typing) an email address, Calc will automatically convert it into a clickable email. However, after that, I can not modify the email anymore. Neither in the forumlar bar, nor in the cell itself. The cursor just jumps to either the start or end of the string.

### Steps to Reproduce

1. Type email address
2. go to another field
3. open email field
4. try to insert a char somewhere in between

### Expected Behavior

-

### Actual Behavior

-

### Screenshots

https://github.com/CollaboraOnline/online/assets/1321720/5039d73e-047f-4d20-b314-65b34c34fb2b

### Desktop

(Please complete the following information)
 - **OS:** macOS
 - **Browser:** Firefox latest
 - **Version:** 115.0.3

### Smartphone

-

### Additional Context

Collabora v23.05.2.2

**Comment by relikd** [comment]
added screen recording

**Comment by Minion3665** [comment]
Yep, I can reproduce this too (I did so on share). Additionally, while the edit link dialog pops up when you press the button to edit the link, editing the text in it does not change the document. The link box is also unfilled at the start and editing the link to another mailto link prepends http:// and breaks the link. 

(Perhaps we should also allow protocols that aren't http/https when handling links?)

**Comment by eszkadev** [comment]
Email address become converted to "link field" so to edit that, you need to click "edit hyperlink" button in the popup.

**Comment by Minion3665** [comment]
Everything apart from
> The link box is also unfilled at the start

(including everything else about the link box not properly working) is also reproducible on chromium 115.0.5790.102

**Comment by Minion3665** [comment]
> Email address become converted to "link field" so to edit that, you need to click "edit hyperlink" button in the popup.

@eszkadev this doesn't quite work, here's a video demonstrating 
[Screencast from 2023-08-10 13-01-02.webm](https://github.com/CollaboraOnline/online/assets/34243578/68452d8e-4e34-48dd-a093-398d647a3d20)

**Comment by relikd** [comment]
@eszkadev The Link editing works for me but the UX is still terrible, namely you'll have to update two fields – instead of just putting your cursor somewhere and insert the missing char.

**Comment by rpear** [comment]
Still broken UI in 24.04.14.2. Same as above about having to edit in 2 places, but also if you attempt to edit the text in the formula bar it can break it in different ways, including the email address completely disappearing.  

Tested in latest Online on a Chromebook.
COOLWSD version: 24.04.14.2snapshot(git hash: acbf1cbeb7 (E))
LOKit version: Collabora Office 24.04.14.20250423(git hash: a9a2a62e81)
Served by: Ubuntu 22.04.5 LTS
Server ID: b2f6405f
WOPI host: nextcloud 30.0.6.3 - richdocuments 8.5.4

**Comment by smehrbrodt** [comment]
Fixed with https://gerrit.collaboraoffice.com/plugins/gitiles/online/+/6af780d54f6bc8a227861fdd4018fde0ba9af1a8%5E%21/

Links will now be converted to normal text when the cell is edited (and converted back to hyperlink when editing is finished).
